src/Processor/Braintree/Subscription/Update/ChangeBillingCycleStrategy.php

```diff
@@ -73,15 +75,71 @@ private function createReplacementSubscription(Subscription $original): Subscrip
             ->setAddOns($addOns)
             ->setDiscounts($discounts);
 
-        // Braintree doesn't allow firstBillingDates in the past.
-        // When upgrading an existing subscription to new billing cycle, clear the startDate if the startDate is in the past.
+        $newSubscription = $this->handleFirstBillingDate($newSubscription);
+        $newSubscription = $this->handleTrialPeriod($newSubscription, $original);
+
+        return $newSubscription;
+    }
+
+    /**
+     * Braintree doesn't allow firstBillingDates in the past.
+     * When upgrading an existing subscription to new billing cycle,
+     * clear the startDate if the startDate is in the past.
+     */
+    private function handleFirstBillingDate(Subscription $subscription): Subscription
+    {
         $today = new \DateTime('utc');
 
-        if ($newSubscription->getStartDate() < $today) {
-            $newSubscription->beginImmediately();
+        if ($subscription->getStartDate() > $today) {
+            return $subscription;
         }
 
-        return $newSubscription;
+        return $subscription->beginImmediately();
+    }
+
+    /**
+     * Preserve the original trial period when a trial subscription changes billing cycles.
+     */
+    private function handleTrialPeriod(Subscription $newSubscription, Subscription $originalSubscription): Subscription
+    {
+        $isInTrial = $originalSubscription->getTrial() instanceof Trial;
+        $hasStartDate = $originalSubscription->getStartDate() instanceof \DateTime;
+
+        if (!$isInTrial || !$hasStartDate) {
+            return $newSubscription;
+        }
+
+        $newTrialDuration = $this->findDaysUntilNextTrialStart($originalSubscription->getStartDate());
+        $newTrial = new Trial($newTrialDuration, TrialUnit::day());
+
+        // Clear the first billing date, as you cannot define
+        // a first billing date & a trial period.
+        $newSubscription->beginImmediately();
+
+        return $newSubscription->setTrial($newTrial);
+    }
+
+    private function findDaysUntilNextTrialStart(?\DateTime $originalStartDate): int
+    {
+        if (!($originalStartDate instanceof \DateTime)) {
+            return 0;
+        }
+
+        $today = new \DateTimeImmutable('UTC');
+        $startOfToday = $today->setTime(0, 0, 0);
+        $trialEndDate = $originalStartDate;
+
+        $difference = $trialEndDate->diff($startOfToday);
+        if (!($difference instanceof \DateInterval)) {
+            return 0;
+        }
+
+        $remainingTrialDays = $difference->days;
+        if (is_bool($remainingTrialDays)) {
```

Comment:
This was a silly thing for static analysis...

Long story short... you can be in a situation where

```php
$validDate = new DateTimeImmutable('utc');
$invalidDate = new DateTimeImmutable('some way to make an invalid date');

$diff = $validDate->difference($invalidDate);

// $diff->days would be false since it can't calculate it.
```

So I needed to confirm it was not false, so that we could use it later.

@jcalebj